website/docs/r/storagegateway_gateway.html.markdown

```diff
@@ -0,0 +1,97 @@
+---
+layout: "aws"
+page_title: "AWS: aws_storagegateway_gateway"
+sidebar_current: "docs-aws-resource-secretsmanager-secret"
+description: |-
+  Manages an AWS Storage Gateway file, tape, or volume gateway in the provider region
+---
+
+# aws_storagegateway_gateway
+
+Manages an AWS Storage Gateway file, tape, or volume gateway in the provider region.
+
+~> NOTE: The Storage Gateway API requires the gateway to be connected to properly return information after activation. If you are receiving `The specified gateway is not connected` errors during resource creation (gateway activation), ensure your gateway instance meets the [Storage Gateway requirements](https://docs.aws.amazon.com/storagegateway/latest/userguide/Requirements.html).
+
+## Example Usage
+
+### File Gateway
+
+```hcl
+resource "aws_storagegateway_gateway" "example" {
+  gateway_ip_address = "1.2.3.4"
+  gateway_name       = "example"
+  gateway_timezone   = "GMT"
+  gateway_type       = "FILE_S3"
+}
+```
+
+### Tape Gateway
+
+```hcl
+resource "aws_storagegateway_gateway" "example" {
+  gateway_ip_address = "1.2.3.4"
+  gateway_name       = "example"
+  gateway_timezone   = "GMT"
+  gateway_type       = "VTL"
+  media_changer_type = "AWS-Gateway-VTL"
+  tape_drive_type    = "IBM-ULT3580-TD5"
+}
+```
+
+### Volume Gateway (Cached)
+
+```hcl
+resource "aws_storagegateway_gateway" "example" {
+  gateway_ip_address = "1.2.3.4"
+  gateway_name       = "example"
+  gateway_timezone   = "GMT"
+  gateway_type       = "CACHED"
+}
+```
+
+### Volume Gateway (Stored)
+
+```hcl
+resource "aws_storagegateway_gateway" "example" {
+  gateway_ip_address = "1.2.3.4"
+  gateway_name       = "example"
+  gateway_timezone   = "GMT"
+  gateway_type       = "STORED"
+}
+```
+
+## Argument Reference
+
+~> **NOTE:** One of `activation_key` or `gateway_ip_address` must be provided for resource creation (gateway activation). Neither is required for resource import. If using `gateway_ip_address`, Terraform must be able to make an HTTP (port 80) GET request to the specified IP address from where it is running.
+
+The following arguments are supported:
+
+* `gateway_name` - (Required) Name of the gateway.
+* `gateway_timezone` - (Required) Time zone for the gateway. The time zone is of the format "GMT-hr:mm" or "GMT+hr:mm". For example, `GMT-4:00` indicates the time is 4 hours behind GMT. The time zone is used, for example, for scheduling snapshots and your gateway's maintenance schedule.
```

Comment:
can this be UTC in addition to GMT? should also mention this can just be `GMT`?

Reply:
When using `UTC` I receive an error from the API 😄  `aws_storagegateway_gateway.test: error activating Storage Gateway Gateway: InvalidGatewayRequestException: The specified request contains invalid parameters.` -- I will add `"GMT"` as a valid option in documentation!

Not asked probably worth noting that given the [API reference](https://docs.aws.amazon.com/storagegateway/latest/APIReference/API_ActivateGateway.html), I would be a little hesitant to implement strict plan time validation for `^GMT` 😅 

> A value that indicates the time zone you want to set for the gateway. The time zone is of the format "GMT-hr:mm" or "GMT+hr:mm". For example, GMT-4:00 indicates the time is 4 hours behind GMT. GMT+2:00 indicates the time is 2 hours ahead of GMT. The time zone is used, for example, for scheduling snapshots and your gateway's maintenance schedule.
>
> Type: String
>
> Length Constraints: Minimum length of 3. Maximum length of 10.